xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/doc/XWikiAttachmentContent.java

```diff
@@ -153,7 +154,32 @@ public void setId(long id)
     @Override
     public Object clone()
     {
-        return new XWikiAttachmentContent(this);
+        return clone(true);
+    }
+
+    /**
+     * Clone current instance and possibly copy the content.
+     * Note that {@link #clone()} never copy the actual content of the instance.
+     * @param skipContent {@code false} to also copy the content while performing the clone.
+     * @return a clone with the copied content or not depending on the parameter.
+     * @since 17.2.0RC1
+     * @since 16.10.5
+     * @since 16.4.7
+     */
+    @Unstable
+    public XWikiAttachmentContent clone(boolean skipContent)
+    {
+        XWikiAttachmentContent clone = new XWikiAttachmentContent(this);
+        // this AttachmentContent is not attached to any document yet.
+        clone.setOwnerDocument(null);
```

Comment:
Note that this is also used for the standard clone(), maybe we need to avoid that?